### PR TITLE
external audit proto spec v2

### PR DESCRIPTION
**Pull request by tobiaszheller**
This PR adds proto spec for service that will be used in external audit/byob feature.
Service will be implemented in enterprise repo.

Documentation is added as part of proto.

Part of: https://github.com/gravitational/cloud/issues/4661
RFD: https://github.com/gravitational/cloud/blob/rfd/77-bring-your-own-bucket/rfd/0077-Bring-your-own-bucket.md

This is 2nd attempt to proto definition based on comments from: https://github.com/gravitational/teleport/pull/32833#issuecomment-1744474263
and internal conversations. 
In short, there are 3 good options to implement it:
1. Multiple `external_cloud_audit` and `single cluster_external_cloud_audit` which references by name
2. Single `external_cloud_audit` and `single cluster_external_cloud_audit` which references by name
3. Single `external_cloud_audit` and `single cluster_external_cloud_audit` which is copy of `external_cloud_audit`.
This PR implements option 3.

Original: https://github.com/gravitational/teleport/pull/31564

### CODE REVIEW

**Review by nklaassen — APPROVED** [review]
Looks good to me, I just suggest renaming the RPCs that deal with the draft configuration to make that more clear.

File: api/proto/teleport/externalcloudaudit/v1/externalcloudaudit.proto [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+message ExternalCloudAudit {
+  // Header is the header for the resource.
+  teleport.header.v1.ResourceHeader header = 1;
+  // Spec is the specification for the external cloud audit.
+  ExternalCloudAuditSpec spec = 2;
+}
+
+// ExternalCloudAuditConfigSpec is the specification of external cloud audit.
+message ExternalCloudAuditSpec {
+  // IntegrationName is name of existing OIDC intagration used to
```
Comment (marcoandredinis) [inline api/proto/teleport/externalcloudaudit/v1/externalcloudaudit.proto:36]: :nail_care: 
The integration subkind is `aws-oidc`

```suggestion
  // IntegrationName is name of existing AWS OIDC intagration used to
```

File: api/proto/teleport/externalcloudaudit/v1/externalcloudaudit_service.proto [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+// test connection and call PromoteToClusterClusterExternalCloudAudit again.
+//
+// Supporting ExternalCloudAudit as resources: "draft" and "cluster", allow us
+// test connection before actually enabling it in cluster.
+service ExternalCloudAuditService {
+  // GetDraftExternalCloudAudit returns the draft external cloud audit configuration resource.
+  rpc GetDraftExternalCloudAudit(GetDraftExternalCloudAuditRequest) returns (GetDraftExternalCloudAuditResponse);
+
+  // UpsertDraftExternalCloudAudit upserts a draft external cloud Audit resource.
+  rpc UpsertDraftExternalCloudAudit(UpsertDraftExternalCloudAuditRequest) returns (UpsertDraftExternalCloudAuditResponse);
```
Comment (marcoandredinis) [inline api/proto/teleport/externalcloudaudit/v1/externalcloudaudit_service.proto:42]: I'm guessing this method always sets the resource state to Draft.

And the Disable method does nothing to the ExternalCloudAudit resource, but does remove the ClusterExternalCloudAudit.

Is that it?

**Review by marcoandredinis — APPROVED** [review]

File: api/proto/teleport/externalcloudaudit/v1/externalcloudaudit_service.proto (hunk 2 above)
Comment (tobiaszheller) [reply api/proto/teleport/externalcloudaudit/v1/externalcloudaudit_service.proto:42]: yes, exactly
